language/extensions/move-table-extension/src/lib.rs

```diff
@@ -49,10 +49,31 @@ impl Display for TableHandle {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct TableInfo {
+    pub key_type: MoveTypeLayout,
+    pub value_type: MoveTypeLayout,
```

Comment:
I think a type tag is what we need..

Reply:
That may be a better choice. Let me look at what that spits out:

https://github.com/aptos-labs/aptos-core/blob/main/api/types/src/convert.rs#L468-L482

Reply:
{TableHandle(294284197927973660676668354886091714884): TableInfo { key_type: U64, value_type: Struct(StructTag { address: 00000000000000000000000000000002, module: Identifier("TableTests"), name: Identifier("Balance"), type_params: [] }) }}
[ PASS    ] 0x2::TableTests::test_remove_fail
{TableHandle(294284197927973660676668354886091714884): TableInfo { key_type: U64, value_type: U64 }}
[ PASS    ] 0x2::TableTests::test_remove_removed
{TableHandle(294284197927973660676668354886091714884): TableInfo { key_type: Address, value_type: Struct(StructTag { address: 00000000000000000000000000000002, module: Identifier("TableTests"), name: Identifier("Balance"), type_params: [] }) }}
[ PASS    ] 0x2::TableTests::test_struct
{TableHandle(294284197927973660676668354886091714884): TableInfo { key_type: Address, value_type: Struct(StructTag { address: 00000000000000000000000000000002, module: Identifier("Table"), name: Identifier("Table"), type_params: [Address, U128] }) }}
{TableHandle(187267989122364641638725157156538820950): TableInfo { key_type: Address, value_type: U128 }, TableHandle(294284197927973660676668354886091714884): TableInfo { key_type: Address, value_type: Struct(StructTag { address: 00000000000000000000000000000002, module: Identifier("Table"), name: Identifier("Table"), type_params: [Address, U128] }) }}
{TableHandle(160937847609835965615946043614408101562): TableInfo { key_type: Address, value_type: U128 }, TableHandle(187267989122364641638725157156538820950): TableInfo { key_type: Address, value_type: U128 }, TableHandle(294284197927973660676668354886091714884): TableInfo { key_type: Address, value_type: Struct(StructTag { address: 00000000000000000000000000000002, module: Identifier("Table"), name: Identifier("Table"), type_params: [Address, U128] }) }}
[ PASS    ] 0x2::TableTests::test_table_of_tables

yeah looks better and this will give us type / field names